pkg/controllers/dynakube/logmonitoring/daemonset/reconciler_test.go

```diff
@@ -295,6 +301,63 @@ func TestGenerateDaemonSet(t *testing.T) {
 
 		assert.Equal(t, daemonset.Spec.Template.Spec.NodeSelector, customNodeSelector)
 	})
+
+	t.Run("generate a daemonset with no kubernetes cluster name set in env and arg section if no MEID and all scopes set", func(t *testing.T) {
+		dk := createDynakube(true)
+		dk.Status.KubernetesClusterMEID = ""
+
+		reconciler := NewReconciler(nil, fake.NewClient(), dk)
+		daemonset, err := reconciler.generateDaemonSet()
+		require.NoError(t, err)
+		require.NotNil(t, daemonset)
+
+		init := daemonset.Spec.Template.Spec.InitContainers[0]
+		require.NotContains(t, init.Args, fmt.Sprintf("-p dt.entity.kubernetes_cluster=$(%s)", entityEnv))
+
+		require.Nil(t, env.FindEnvVar(init.Env, entityEnv))
+	})
+
+	t.Run("both scopes set, MEID missing - wait, DS not created", func(t *testing.T) {
+		dk := createDynakube(true)
+		dk.Status.KubernetesClusterMEID = ""
+		setScopes(dk, true, true)
```

Comment:
```suggestion
		setScopes(dk, true, true)
```
TBH i prefer to no use this pattern, cause it's super unclear what does it mean `setScopes(true, true)`
Instead, just create 2 similar test functions, don't forget to mark them as t.Helper and 
- setReadScopes
- setWriteScopes

Reply:
fair enough, done https://github.com/Dynatrace/dynatrace-operator/pull/5239/commits/3b451aaf990ee0ad0fb0949713a390a932305d32